common/namespace/handler.go

```diff
@@ -787,3 +781,65 @@ func (d *HandlerImpl) validateVisibilityArchivalURI(URIString string) error {
 
 	return archiver.ValidateURI(URI)
 }
+
+// validateRetentionDuration ensures that retention duration can't be set below a sane minimum.
+func validateRetentionDuration(retention time.Duration, isGlobalNamespace bool) error {
+	min := MinRetentionLocal
+	max := common.MaxWorkflowRetentionPeriod
+	if isGlobalNamespace {
+		min = MinRetentionGlobal
+	}
+	if retention < min || retention > max {
+		return errInvalidRetentionPeriod
+	}
+	return nil
+}
+
+func validateStateUpdate(oldNamespace *persistence.GetNamespaceResponse, newNamespace *workflowservice.UpdateNamespaceRequest) error {
```

Comment:
i would propose
* splitting namespace mutation API into 2, one for the namespace configs, one for replication related metadata
* use set / put API for namespace mutation (instead of update, plus update itself is misleading, e.g. how to specify a bad binary should be deleted, easily?)